src/models/SearchResult.kt

```diff
@@ -1,19 +1,5 @@
 package com.rafael.models
 
-class SearchResult(val endUsers: List<EndUser>) {
-    fun result(): EndUser? {
-        return if (oneEndUserFound()) {
-            endUsers.first()
-        } else {
-            null
-        }
-    }
-
-    fun hasResult(): Boolean {
-        return oneEndUserFound()
-    }
-
-    private fun oneEndUserFound(): Boolean {
-        return endUsers.count() == 1
-    }
+class SearchResult(private val endUsers: List<EndUser>) {
+    fun uniqueResult() = endUsers.firstOrNull()
```

Comment:
you probably break this rule, we need to make sure to have only one end user to have a result